packages/aws-cdk/lib/api/nested-stack-helpers.ts

```diff
@@ -0,0 +1,142 @@
+import * as path from 'path';
+import * as cxapi from '@aws-cdk/cx-api';
+import * as fs from 'fs-extra';
+import { debug } from '../logging';
+import { ISDK } from './aws-auth';
+import { LazyListStackResources, ListStackResources } from './evaluate-cloudformation-template';
+import { CloudFormationStack, Template } from './util/cloudformation';
+
+/**
+ * Reads the currently deployed template from CloudFormation and adds a
+ * property, `NestedTemplate`, to any nested stacks that appear in either
+ * the deployed template or the just synthesized template. `NestedTemplate`
+ * is populated with contents of the nested template. This is done for all
+ * nested stack resources to arbitrary depths.
+ */
+export async function loadCurrentTemplateWithNestedStacks(
+  rootStackArtifact: cxapi.CloudFormationStackArtifact, sdk: ISDK,
+): Promise<TemplateWithNestedStackNames> {
+  const deployedTemplate = await loadCurrentTemplate(rootStackArtifact, sdk);
+  const nestedStackNames = await addNestedTemplatesToGeneratedAndDeployedStacks(rootStackArtifact, sdk, {
+    generatedTemplate: rootStackArtifact.template,
+    deployedTemplate: deployedTemplate,
+    deployedStackName: rootStackArtifact.stackName,
+  });
+
+  return {
+    deployedTemplate,
+    nestedStackNames,
+  };
+}
+
+/**
+ * Returns the currently deployed template from CloudFormation that corresponds to `stackArtifact`.
+ */
+export async function loadCurrentTemplate(stackArtifact: cxapi.CloudFormationStackArtifact, sdk: ISDK): Promise<Template> {
+  debug(`Reading existing template for stack ${stackArtifact.displayName}.`);
```

Comment:
I don't think we need this `debug()` statement...?

It's already in `CloudFormationDeployments.readCurrentTemplate()` (and I agree it should live there), which calls this method, which means this would be displayed twice.